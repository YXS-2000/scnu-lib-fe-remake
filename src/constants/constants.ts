// this will passed by backend
export const activityTags = ['阅读马拉松', '旅游分享会', '周五影院',
  '观影沙龙', '喜阅读书会', '诗会'];

