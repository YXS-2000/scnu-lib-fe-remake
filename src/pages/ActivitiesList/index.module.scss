.filterBar {
  width: 100%;
  max-height: 150px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 20px;
}