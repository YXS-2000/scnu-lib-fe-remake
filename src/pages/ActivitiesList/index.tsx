export default function ActivitiesList() {
  return <div>this will be used to show activities</div>;
}
