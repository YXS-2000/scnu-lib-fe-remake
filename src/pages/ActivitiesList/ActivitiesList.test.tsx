import { render, RenderResult, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { activityTags } from '../../constants/constants';
import ActivitiesList from './index';

describe('<ActivitiesList />', () => {
  describe('Logical testing', () => {
    let root: RenderResult;

    beforeEach(() => {
      root = render(<ActivitiesList />);
    });

    describe('<ActivityFilter />', () => {
      it('should show initial selector and filter tags', () => {
        screen.getByText('标签筛选：');
        activityTags.map(tag => screen.getAllByText(tag.name)[0]);
      });
    
      it('should show tag in filter when tag selected', () => {
        const unselectedFilter = screen.queryByTestId('selected-filter');
        const selector = screen.getAllByText(activityTags[0].name)[0];
    
        userEvent.click(selector);
    
        const selectedFilter = screen.getByTestId('selected-filter');
    
        expect(unselectedFilter).toBeNull();
        expect(selectedFilter).toBeDefined();
        expect(selectedFilter.textContent).toBe(activityTags[0].name);
      });
    
      it('should show selected once when double click', () => {
        
        const selector = screen.getAllByText(activityTags[0].name)[0];
    
        userEvent.dblClick(selector);
    
        const selectedFilter = screen.queryAllByTestId('selected-filter');
    
        expect(selectedFilter).toHaveLength(1);
      });
    
      it('should dispear when click close', () => {
        const { container } = root;
        const selector = screen.getAllByText(activityTags[0].name)[0];
    
        userEvent.click(selector);
    
        const selectedFilter = screen.queryByTestId('selected-filter');
        const selectedFilterCancelButton = container.getElementsByClassName('semi-icon-close');
    
        userEvent.click(selectedFilterCancelButton.item(0) as Element);
    
        const selectedFilterAfterClick = screen.queryByTestId('selected-filter');
    
        expect(selectedFilter).toBeDefined();
        expect(selectedFilterAfterClick).toBeNull();
      });  
    });  

    describe('<ActivitySearchFilter />', () => {
      it('should render search bar with placehold and search button', () => {
        screen.getByPlaceholderText('输入标题搜索');
        const searchButton = screen.getByRole('button', {
          name: '搜索按钮',
        });
        expect(searchButton).toHaveTextContent('搜索');
      });

      it('should change input value when type title', () => {
        const input = screen.getByPlaceholderText('输入标题搜索');
        
        userEvent.type(input, 'title');

        expect((input.closest('input') as HTMLInputElement).value).toEqual('title');
      });
    });
  });

  describe('UI testing', () => {
    it('should render initial UI correctly', () => {
      const tree = renderer
        .create(<ActivitiesList />)
        .toJSON();
      
      expect(tree).toMatchSnapshot();
    });
  });
});
