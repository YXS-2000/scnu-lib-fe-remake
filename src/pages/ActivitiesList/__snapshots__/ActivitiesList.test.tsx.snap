// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActivitiesList /> UI testing should render initial UI correctly 1`] = `
<div>
  <div
    className="filter"
  >
    标签筛选：
    <div
      className="semi-space filterTags semi-space-align-center semi-space-horizontal semi-space-tight-horizontal semi-space-tight-vertical"
      style={Object {}}
    >
      <div
        className="semi-space unselectedTags semi-space-align-center semi-space-horizontal semi-space-tight-horizontal semi-space-tight-vertical"
        style={Object {}}
      >
        <div
          aria-label="Tag: 阅读马拉松"
          className="semi-tag semi-tag-default semi-tag-light semi-tag-blue-light"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          style={Object {}}
          tabIndex={0}
        >
          <div
            className="semi-tag-content"
          >
            阅读马拉松
          </div>
        </div>
        <div
          aria-label="Tag: 旅游分享会"
          className="semi-tag semi-tag-default semi-tag-light semi-tag-red-light"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          style={Object {}}
          tabIndex={0}
        >
          <div
            className="semi-tag-content"
          >
            旅游分享会
          </div>
        </div>
        <div
          aria-label="Tag: 周五影院"
          className="semi-tag semi-tag-default semi-tag-light semi-tag-green-light"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          style={Object {}}
          tabIndex={0}
        >
          <div
            className="semi-tag-content"
          >
            周五影院
          </div>
        </div>
        <div
          aria-label="Tag: 观影沙龙"
          className="semi-tag semi-tag-default semi-tag-light semi-tag-yellow-light"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          style={Object {}}
          tabIndex={0}
        >
          <div
            className="semi-tag-content"
          >
            观影沙龙
          </div>
        </div>
        <div
          aria-label="Tag: 喜阅读书会"
          className="semi-tag semi-tag-default semi-tag-light semi-tag-teal-light"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          style={Object {}}
          tabIndex={0}
        >
          <div
            className="semi-tag-content"
          >
            喜阅读书会
          </div>
        </div>
        <div
          aria-label="Tag: 诗会"
          className="semi-tag semi-tag-default semi-tag-light semi-tag-violet-light"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          style={Object {}}
          tabIndex={0}
        >
          <div
            className="semi-tag-content"
          >
            诗会
          </div>
        </div>
      </div>
      <div
        className="semi-space selectedTags semi-space-align-center semi-space-horizontal semi-space-tight-horizontal semi-space-tight-vertical"
        style={Object {}}
      />
    </div>
  </div>
</div>
`;
